controllers/provisioning_controller.go

```diff
@@ -178,7 +179,13 @@ func (r *ProvisioningReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error
 		return ctrl.Result{}, err
 	}
 
-	info := r.provisioningInfo(baremetalConfig, &containerImages)
+	// Get cluster-wide proxy information
+	clusterWideProxy, err := r.OSClient.ConfigV1().Proxies().Get(context.Background(), "cluster", metav1.GetOptions{})
```

Comment:
Should the operator watch for the Proxy config and react if it changes day -2 ? (it seems unlikely that a customer might change their proxy config on day-2) If so, do we need to add it to the list of watched resources here https://github.com/openshift/cluster-baremetal-operator/blob/master/controllers/provisioning_controller.go#L401 ?

Reply:
Yes that makes sense, I was trying to figure out how to do this actually, thanks!

Reply:
@sadasu How do I add this as being watched? It seems like I have a few options. I could use `For()`, but is that enough? Or do I need a `Watches()` with some specific handler that knows what to do?

Reply:
We could probably use Watches() from the controller-runtim https://github.com/openshift/cluster-baremetal-operator/blob/aab97dfe194c9f527f54209bb530ae249cedead3/vendor/sigs.k8s.io/controller-runtime/pkg/builder/controller.go#L115 to start watching the Proxies{} resource.

Reply:
Do you know what the eventhandler would be?

Reply:
I tested with `Owns` instead and my metal3 pod did eventually get restarted after modifying the Proxy object.